### ISSUE TITLE
Python Model with BLS: failed to get cuda pointer device attribute

### ISSUE THREAD

**Issue by naor2013** [triton-inference-server/server#4215]
**Description**
I'm using a Python model with BLS to call a TorchScript model that runs on a gpu.
I can see that when calling the Python model, it sends the request to the TorchScript model and succeeds based on the "add response output" log that happens after the model call.
Then I get a "failed to get cuda pointer device attribute: initialization error".
I tried running the Python model with kind_gpu and kind_cpu, with and without the “ FORCE_CPU_ONLY_INPUT_TENSORS” parameter.

**Triton Information**
What version of Triton are you using?
I’m using 22.02. Is there any chance that this bug relates to the many pull requests to the Python backend in the last month, specifically [this one?](https://github.com/triton-inference-server/python_backend/pull/126) 

Are you using the Triton container or did you build it yourself?
Using the Triton container.

**Comment by Tabrizian** [comment]
Hi @naor2013! Looks like call to [this function](https://github.com/triton-inference-server/python_backend/blob/c9f79ce25491a66075a563c655a9cc3285069cfb/src/pb_utils.cc#L85) failed. 

* Have you started the container with `--gpus all`? It looks like there is some problem with CUDA initialization.
* What is the CUDA Driver version installed on your system? What GPU are you using is it a datacenter GPU?

The commit that you have shared is not included in the 22.02 release and will be first included in the 22.04 release.

**Comment by naor2013** [comment]
> I think there might some issue in the CUDA driver installation. Can you share the output of `nvidia-smi`?
> 
> The pull request you linked I think is unrelated to this issue. This fixes a bug when using CUDA shared memory in the input with python backend.

I'm using the compatibility driver feature, so my actual Cuda is 11.0 but nvidia-smi shows Cuda version 11.6.

What exactly do you want to see in nvidia-smi? We have many GPUs all set up and used frequently, so it'll be very weird if that is a Cuda driver installation issue.

I tried to convert the output tensor to DLPack and from DLPack to torch, I got an issue from torch: "Specified device cuda:0 does not match device of data cuda:-2"
Does that help to narrow down what the issue is?

**Comment by Tabrizian** [comment]
Yes, that is very helpful. Can you try running the 22.03 release? It includes this commit that could be related: https://github.com/triton-inference-server/python_backend/commit/56727b5b78d5d04720eac718c5f3ef1b9d591a94

**Comment by naor2013** [comment]
> Yes, that is very helpful. Can you try running the 22.03 release? It includes this commit that could be related: [triton-inference-server/python_backend@56727b5](https://github.com/triton-inference-server/python_backend/commit/56727b5b78d5d04720eac718c5f3ef1b9d591a94)

It looks like it got merged in early February, does that not mean it is included in 22.02?

**Comment by Tabrizian** [comment]
My bad. You are correct. It is included in the 22.02 release. I'm not sure what else could be wrong. Can you share the contents of your model repository along with the client so that we can look into this?

**Comment by naor2013** [comment]
@Tabrizian I'll try to reproduce it in a new environment so I'll have clear steps to reproduce.

I have another unrelated issue but it might have a quick answer (if not, I'll open a different issue).
I'm trying to use the Python backend to call my Pytorch model directly (I'm testing something specific, which is why I currently don't use the Pytorch backend).

When I test it with perf analyzer, using low concurrency (1-2) it works fine, but with higher concurrency (3+), it fails with "Dtype is invalid".

What can I try? The input is coming from a json file to perf analyzer and the output is always the same type (I added a check for it), so what can it be?

**Comment by naor2013** [comment]
@Tabrizian I don't store them, the only thing I store is the model object in the initialize function.

**Comment by Tabrizian** [comment]
Can you open a new issue with the reproducing instructions so that we can look into this?
